compact_str/src/repr/mod.rs

```diff
@@ -108,6 +108,23 @@ impl Repr {
         }
     }
 
+    #[inline]
+    pub fn from_fmt(val: impl fmt::Display) -> Self {
+        let len = crate::count!("{}", val);
+
+        if len == 0 {
+            EMPTY
+        } else if len <= MAX_SIZE {
+            let mut buffer = [0_u8; MAX_SIZE];
+            let text = format_into_buffer(&mut buffer, &val);
+            Self::new(text)
```

Comment:
We should either `impl Write` for `Repr` or have this method return a `CompactStr`, for two reasons:
1. Writing into a buffer and then using that buffer to create a `Repr` results in an additional copy
2. Using something like `format_into_buffer(...)` is another part of code that manually writing bytes and ideally we have as few of those as possible, since these are the easiest things to get wrong

Reply:
I have fixed this performance issue by implementing `InlineString::from_fmt` and call the `format_into_buffer` directly on the buffer used by `InlineString`.

Reply:
I have now refactored the code and inline `format_into_buffer` into `InlineString::from_fmt`.

Since `format_into_buffer` simply uses `std::io::Cursor`, it's unlikely to have any bugs.

Reply:
I have pushed another commit to simplify `InlineString::from_fmt`, by removing use of `std::io::Cursor`.